[MacOS CI] Issue with cmath, boost and Apple

<!--- WARNING: This is an issue tracker. Before opening a new issue make sure you read https://github.com/PointCloudLibrary/pcl/blob/master/CONTRIBUTING.md#using-the-issue-tracker. -->

<!--- Provide a general summary of the issue in the Title above -->

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Operating System and version: MacOS 10.14 (CI)
* Compiler: AppleClang 11
* PCL Version: HEAD

## Context
MacOS CI is red

## Current Behavior
```
/usr/local/include/boost/math/special_functions/fpclassify.hpp:552:17: error: no member named 'isnan' in namespace 'std'; did you mean simply 'isnan'?
        return (std::isnan)(x);
                ^~~~~
/usr/local/include/boost/math/special_functions/math_fwd.hpp:885:9: note: 'isnan' declared here
   bool isnan BOOST_NO_MACRO_EXPAND(T t);
        ^
```

--- a/.ci/azure-pipelines/build-macos.yml
+++ b/.ci/azure-pipelines/build-macos.yml
@@ -22,10 +22,9 @@ jobs:
           cd $GOOGLE_TEST_DIR && git checkout release-1.8.1
         displayName: 'Install Dependencies'
       - script: |
-          XCODE_APP_PATH=`ls -d /Applications/Xcode*`
           mkdir $BUILD_DIR && cd $BUILD_DIR
           cmake $(Build.SourcesDirectory) \
-            -DCMAKE_OSX_SYSROOT="$XCODE_APP_PATH/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk" \
+            -DCMAKE_OSX_SYSROOT="/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk" \
             -DCMAKE_CXX_FLAGS="$CMAKE_CXX_FLAGS" \
             -DGTEST_SRC_DIR="$GOOGLE_TEST_DIR/googletest" \
             -DGTEST_INCLUDE_DIR="$GOOGLE_TEST_DIR/googletest/include" \

--- a/.ci/azure-pipelines/build-macos.yml
+++ b/.ci/azure-pipelines/build-macos.yml
@@ -2,8 +2,8 @@ jobs:
   - job: osx
     strategy:
       matrix:
-        # macOS Catalina:
-        #   VMIMAGE: 'macOS-10.15'
+        macOS Catalina:
+          VMIMAGE: 'macOS-10.15'
         macOS Mojave:
           VMIMAGE: 'macOS-10.14'
     timeoutInMinutes: 0

--- a/.ci/azure-pipelines/build-macos.yml
+++ b/.ci/azure-pipelines/build-macos.yml
@@ -22,8 +22,10 @@ jobs:
           cd $GOOGLE_TEST_DIR && git checkout release-1.8.1
         displayName: 'Install Dependencies'
       - script: |
+          XCODE_APP_PATH=`ls -d /Applications/XCode*`
           mkdir $BUILD_DIR && cd $BUILD_DIR
           cmake $(Build.SourcesDirectory) \
+            -DCMAKE_OSX_SYSROOT="$XCODE_APP_PATH/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk" \
             -DCMAKE_CXX_FLAGS="$CMAKE_CXX_FLAGS" \
             -DGTEST_SRC_DIR="$GOOGLE_TEST_DIR/googletest" \
             -DGTEST_INCLUDE_DIR="$GOOGLE_TEST_DIR/googletest/include" \

--- a/.ci/azure-pipelines/build-macos.yml
+++ b/.ci/azure-pipelines/build-macos.yml
@@ -22,7 +22,7 @@ jobs:
           cd $GOOGLE_TEST_DIR && git checkout release-1.8.1
         displayName: 'Install Dependencies'
       - script: |
-          XCODE_APP_PATH=`ls -d /Applications/XCode*`
+          XCODE_APP_PATH=`ls -d /Applications/Xcode*`
           mkdir $BUILD_DIR && cd $BUILD_DIR
           cmake $(Build.SourcesDirectory) \
             -DCMAKE_OSX_SYSROOT="$XCODE_APP_PATH/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk" \

